src/snowflake/cli/api/commands/snow_typer.py

```diff
@@ -136,6 +136,10 @@ def process_result(result):
         # We should improve it in future.
         if not result:
             return
+        if isinstance(result, Generator):
+            for r in result:
+                SnowTyper.process_result(r)
+            return
```

Comment:
How different is this from returning `CollectionResult(generator)`?

Reply:
A CollectionResult is printed all at once via `list(o.result)`, which won't work for `tail -f` style streaming output. What I want is for lines to be printed as soon as they're generated.

I'm open to editing the result handling code to iterate over the elements of a CollectionResult one at a time instead, but I feel like this approach is simpler since it just requires commands to change `return` to `yield`.

Reply:
Hm, would it make sense to make `StreamingResult`? This is a solution we were thinking about for spcs logs streaming. 

The reason I'm concern is inconsistency of commands implementation (return or yield). Hopefully adding `StreamingResult` won't be too complex. WDYT?

Reply:
discussed on Slack, updated to return a `StreamResult`